uqcsbot/scripts/hoogle.py

```diff
@@ -0,0 +1,57 @@
+from uqcsbot import bot, Command
+import requests
+import json
+import html
+
+def get_endpoint(type_sig: str) -> str:
+    unescaped = html.unescape(type_sig)
+
+    return "https://www.haskell.org/hoogle/?mode=json&hoogle=" + unescaped + "&start=0&count=10"
+
+def pretty_hoogle_result(result: dict, is_verbose: bool) -> str:
+    url = result['location']
+    type_sig = result['self']
+    docs = result['docs']
+
+    if is_verbose:
+        return f"`{type_sig}` <{url}|link> {docs}"
+    else:
+        return f"`{type_sig}` <{url}|link>"
+        
+@bot.on_command("hoogle")
+def handle_hoogle(command: Command):
+    '''
+    `!hoogle [--verbose] <TYPE_SIGNATURE>` - Queries the Hoogle Haskell API search engine, 
+    searching Haskell libraries by either function name, or by approximate type signature.
+    '''
+    command_args = command.arg.split() if command.has_arg() else []
+
+    verbose = False
+
+    if '--verbose' in command_args:
+        command_args.remove('--verbose')
+        verbose = True
+
+    if len(command_args) == 0:
+        bot.post_message(command.channel, handle_hoogle.__doc__)
```

Comment:
I like this, good use of `__doc__` 👍 i reckon we'll adopt something similar for other scripts
maybe prepend with `usage: ` just to be 100% clear?